app/actions/ControlActions.js

```diff
@@ -1067,35 +1087,51 @@ export const LOCKACCOUNT_SUCCESS = "LOCKACCOUNT_SUCCESS";
 // account in case of success or error, if leaveUnlock is not informed.
 export const unlockAcctAndExecFn = (
   passphrase,
-  acctNumber,
+  accts,
   fn,
   leaveUnlock
 ) => async (dispatch, getState) => {
   let res = null;
   let fnError = null;
   dispatch({ type: UNLOCKACCOUNT_ATTEMPT });
+  const walletService = sel.walletService(getState());
 
   // sanity checks
   const accounts = sel.balances(getState());
-
-  const account = accounts.find((acct) => acct.accountNumber === acctNumber);
-  if (!account) {
-    throw "Account not found";
-  }
-  if (!account.encrypted) {
-    throw "Account not encrypted";
-  }
+  // do not allow locking of the dex account, as it isn't supposed to lock.
+  const dexAccountName = sel.dexAccount(getState());
+  const dexAccount = accounts.find(
+    (acct) => acct.accountName === dexAccountName
+  );
+  accts.map((acctNum) => {
+    const account = accounts.find((acct) => acct.accountNumber === acctNum);
+    if (!account) {
+      throw "Account not found";
+    }
+    if (!account.encrypted) {
+      throw "Account not encrypted";
+    }
+  });
 
   // unlock wallet
   try {
-    await wallet.unlockAccount(
-      sel.walletService(getState()),
-      passphrase,
-      acctNumber
+    await Promise.all(
+      accts.map(async (acctNumber) => {
+        await wallet.unlockAccount(walletService, passphrase, acctNumber);
+      })
     );
     dispatch({ type: UNLOCKACCOUNT_SUCCESS });
   } catch (error) {
-    // no need to lock as unlock errored.
+    if (String(error).indexOf("invalid passphrase") < 0) {
```

Comment:
I don't think this check is needed: you probably want to re-lock irrespective of the error that happened.

This can't (easily) happen right now, but consider if you're unlocking multiple accounts and one of them has a different passphrase: you'll end up not re-locking the accounts due to this check. This situation could happen in the future if we allow individual passphrases or if the user manually interacted with dcrwallet to set individual pwds.